### PR TITLE
Added support for JamesServerErrors, which contain the http status code and response body

### PR DESCRIPTION
**Pull request by samiulsami**

### CODE REVIEW

File: james/accounts.go [hunk 1]
```diff
@@ -32,14 +32,17 @@ func (js *Service) createAccount(object Object) error {
 	r, err := js.Client.UsersAPI.UpsertUser(context.TODO(), objectAddr).
 		UpsertUserRequest(openapi.UpsertUserRequest{Password: RandomPassword}).Execute()
 
-	if err != nil {
-		return fmt.Errorf("server error: %v", err)
+	if r == nil || err != nil {
```
Comment (Imtiaz246) [inline james/accounts.go]: Don't need to check `r` is nil or not. Just pass `r and err`. Do it for others

File: james/accounts.go [hunk 2]
```diff
@@ -141,11 +153,15 @@ func (js *Service) createGroup(object Object) error {
 		return err
 	}
 	r, err := js.Client.AddressGroupAPI.CreateGroup(context.Background(), grpAddr).Execute()
-	if err != nil {
-		return fmt.Errorf("server error: %v", err)
+	if r == nil || err != nil {
+		return unmarshalToJamesServerError(r)
 	}
+
 	if r.StatusCode >= 300 {
```
Comment (Imtiaz246) [inline james/accounts.go]: we won't check this status >= 300 if already james check and returned error

File: james/errors.go [hunk 3]
```diff
@@ -16,30 +15,33 @@ type JamesServerError struct {
...
 }
 
+func (e JamesServerError) Error() string {
+	return fmt.Sprintf("james server responded with status code: %d, type: %s, message: %s, details: %s\n", e.StatusCode, e.Type, e.Message, e.Details)
+}
+
 func unmarshalToJamesServerError(r *http.Response) *JamesServerError {
-	bodyData, err := io.ReadAll(r.Body)
-	if err != nil {
+	if r == nil {
```
Comment (Imtiaz246) [inline james/errors.go]: Get err and r also. And check if r is nil if nil then we make the struct from the err otherwise we use err.

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

File: james/accounts.go (hunk 1 above)
Comment (samiulsami) [reply james/accounts.go]: fixed

File: james/errors.go (hunk 3 above)
Comment (samiulsami) [reply james/errors.go]: done

File: james/accounts.go (hunk 2 above)
Comment (samiulsami) [reply james/accounts.go]: fixed

**Pull request merged**